dashboard/app/models/shared_blockly_function.rb

```diff
@@ -55,7 +55,7 @@ def to_xml_doc
         end
         xml.title(name, name: 'NAME')
         xml.statement(name: 'STACK') do
-          xml << stack
+          xml << stack.gsub(/\n */, '')
```

Comment:
Interesting - I'd have used a completely different regexp: `/^\s+/`. Yours matches every newline and does a no-op replacement when there's no spaces, and it wouldn't catch other whitespace characters like tabs (not sure if that's relevant here though).